### PR TITLE
[BUG]: Fix bug with pre epoch normalization

### LINKED ISSUE #36294: BUG:unexpected behavior of pandas 1.1.1 dt.normalize() on pre-epoch dates
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
print(pd.__version__)
1.1.1

df = pd.DataFrame({'year': [1969, 2016],'month': [1, 1],'day': [1, 1],'hour': [9, 9]})
df=pd.to_datetime(df)

df

Out[10]: 
0   1969-01-01 09:00:00
1   2016-01-01 09:00:00
dtype: datetime64[ns]

df.dt.normalize()

Out[11]: 
0   1969-01-02
1   2016-01-01
dtype: datetime64[ns]
```

#### Problem description

I would expect pre-epoch behavior to be the same as post-epoch

#### Expected Output
```
Out[10]: 
0   1969-01-01 09:00:00
1   2016-01-01 09:00:00
dtype: datetime64[ns]

df.dt.normalize()

Out[11]: 
0   1969-01-01
1   2016-01-01
dtype: datetime64[ns]
```

#### Output of ``pd.show_versions()``

<details>
commit           : f2ca0a2665b2d169c97de87b8e778dbed86aea07
python           : 3.6.10.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.17763
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : None.None

pandas           : 1.1.1
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.2
setuptools       : 49.6.0.post20200814
Cython           : 0.29.21
pytest           : 6.0.1
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : 1.3.3
lxml.etree       : 4.5.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.16.1
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.8.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : 1.3.19
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.50.1

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #36294
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

The fix propsed by @jbrockmendel worked.

Hope the test is in the right place

### CODE REVIEW

File: doc/source/whatsnew/v1.1.3.rst [hunk 1]
```diff
@@ -36,6 +36,7 @@ Fixed regressions
 - Fixed regression in :meth:`read_excel` with ``engine="odf"`` caused ``UnboundLocalError`` in some cases where cells had nested child nodes (:issue:`36122`,:issue:`35802`)
 - Fixed regression in :class:`DataFrame` and :class:`Series` comparisons between numeric arrays and strings (:issue:`35700`,:issue:`36377`)
 - Fixed regression when setting empty :class:`DataFrame` column to a :class:`Series` in preserving name of index in frame (:issue:`36527`)
+- Fixed regression in :meth:`Series.dt.normalize()` when normalizing pre-epoch dates the result was shifted one day (:issue:`36294`)
```
Comment (dsaxton) [inline doc/source/whatsnew/v1.1.3.rst:39]:
```suggestion
- Fixed regression in :meth:`Series.dt.normalize` when normalizing pre-epoch dates the result was shifted one day (:issue:`36294`)
```

File: pandas/tests/series/test_datetime_values.py [hunk 2]
```diff
@@ -702,3 +702,14 @@ def test_week_and_weekofyear_are_deprecated():
...
         series.dt.weekofyear
+
+
+def test_normalize_pre_epoch_dates():
+    # GH: 36294
+    df = pd.DataFrame(
+        {"year": [1969, 2016], "month": [1, 1], "day": [1, 1], "hour": [9, 9]}
+    )
+    df = pd.to_datetime(df)
+    result = df.dt.normalize()
```
Comment (dsaxton) [inline pandas/tests/series/test_datetime_values.py:713]: Could even be a tad more minimal / easier to follow with e.g.,
```python
s = pd.to_datetime(pd.Series(["1969-01-01 09:00:00", "2016-01-01 09:00:00"]))
result = s.dt.normalize()
```

Comment (phofl) [reply pandas/tests/series/test_datetime_values.py]: Thanks, that is better :)

File: doc/source/whatsnew/v1.1.3.rst (hunk 1 above)
Comment (phofl) [reply doc/source/whatsnew/v1.1.3.rst]: Done

**Review by dsaxton — APPROVED** [review]
Thanks @phofl

**Review by simonjayhawkins — APPROVED** [review]
Thanks @phofl generally lgtm

> Hope the test is in the right place

That seems a reasonable location for the issue code sample.

but the MWE would be `Timestamp("1969-01-01 09:00:00").normalize()`

maybe add that somewhere in pandas\tests\scalar\timestamp\test_timestamp.py

**Comment by phofl** [comment]
@simonjayhawkins  Added the test

**Review by simonjayhawkins — APPROVED** [review]

File: pandas/tests/scalar/timestamp/test_timestamp.py [hunk 3]
```diff
@@ -564,3 +564,10 @@ def test_dt_subclass_add_timedelta(lh, rh):
     result = lh + rh
     expected = SubDatetime(2000, 1, 1, 1)
     assert result == expected
+
+
+def test_normalize_pre_epoch_dates():
```
Comment (jbrockmendel) [inline pandas/tests/scalar/timestamp/test_timestamp.py:569]: in timestamp.test_unary_ops there is a test_normalize this can go right below

Comment (phofl) [reply pandas/tests/scalar/timestamp/test_timestamp.py]: Thanks, moved the test

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**